docs/source/guide/developer/2-jit-vmap-toturial-xbq.md

```diff
@@ -0,0 +1,248 @@
+    
+    # This function will be automatically JIT
+    def automatically_JIT_func1(self, x: torch.Tensor) -> torch.Tensor:
+        pass
+
+    # Use `torch.jit.ignore` to disable JIT and leave this function as Python callback
+    @torch.jit.ignore
+    def not_automatically_JIT__func2(self, x: torch.Tensor) -> torch.Tensor:
+        # you can implement pure Python logic here
+        pass
+
+    # JIT functions can invoke other JIT functions as well as non-JIT functions
+    def automatically_JIT_func3(self, x: torch.Tensor) -> torch.Tensor:
+        y = self.func1(x)
+        z = self.func2(x)
+        pass
+    
+    ...
+    
+```
+
+### Invocation of external vmapped methods inside the subclass
+
+Inside the subclass, external vmapped methods can be invocated by the class methods to be JIT:
+
+```Python
+# Import Pytorch
+import torch
+
+# Import the ModuleBase class from EvoX
+from evox.core import ModuleBase, jit, vmap
+
+
+# One example of the JIT vmapped method defined outside the module
+@jit
+def external_vmap_func(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+    
+    # The method to be vmapped
+    def external_func(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+    	return x + y
+    
+    return vmap(external_func, in_dims=1, out_dims=1, trace=False)(x, y)
+
+# Set an module inherited from the ModuleBase class
+class ExampleModule(ModuleBase):
+    
+    ...    
+    
+    # The internal class method to be JIT   
+    @jit
+    def jit_func(self, p: torch.Tensor) -> torch.Tensor:
+        return external_vmap_func(p, p)
+    
+    ...
+    
+```
+
+```{note}
+If method A invokes vmapped method B, then A and all methods invoke method A can not be vmapped again.
+```
+
+### Internal vmapped methods inside the subclass
+
+Inside the subclass, internal vmapped methods can be  JIT by using the [`trace_impl`](#evox.core.module.trace_impl):
+
+```Python
+# Import Pytorch
+import torch
+
+# Import the ModuleBase class from EvoX
+from evox.core import ModuleBase, jit, vmap, trace_impl
+
+# Set an module inherited from the ModuleBase class
+class ExampleModule(ModuleBase):
+    
+    ...    
+    
+    # The internal vmapped class method to be JIT   
+    @jit
+    def jit_vamp_func(self, p: torch.Tensor) -> torch.Tensor:
+        
+        # The original method
+        # We can not vmap it
+    	def func(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+    		return x+y
+        
+        
+        # The method to be vmapped
+        # We need to use trace_impl to rewrite the original method
+        @trace_impl(func)
+    	def trace_func(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+    		pass
+        
+        return vmap(func, in_dims=1, out_dims=1, trace=False)(p, p)
+    
+    ...
+    
+```
+
+```{note}
+If a class method use [`trace_impl`](#evox.core.module.trace_impl),it will be only available in the trace mode. More details about `[`trace_impl`](#evox.core.module.trace_impl) will be shown in the next part.
```

Comment:
You don't need that many references in the same place. :)
```suggestion
If a class method uses [`trace_impl`](#evox.core.module.trace_impl), it will be only available in the trace mode. More details about trace_impl will be provided in the next section.
```